packages/core/src/impl/WriteApiImpl.ts

```diff
@@ -236,8 +236,13 @@ export default class WriteApiImpl implements WriteApi {
             reject(error)
           },
           complete(): void {
+            // InfluxDB v3 returns 201 for partial success
             // older implementations of transport do not report status code
-            if (responseStatusCode == 204 || responseStatusCode == undefined) {
+            if (
```

Comment:
Is it worth handling all 2XX status codes here, in case this changes in the future?

Reply:
Yes! But! Read this and tell me what you think:
https://github.com/influxdata/influxdb-client-js/issues/263

Reply:
My suggestion is to modify the behavior of the clients as follows: v2 clients will accept `204` and `201` HTTP status codes, and v3 clients will accept any `2xx` status codes. I've updated our backlog item accordingly and included a link to #263 - https://github.com/orgs/influxdata/projects/108/views/1?pane=issue&itemId=73988042.

Reply:
We'll also want to make the response body of a 201 partial write accessible to the caller, whether through a log or some other way I don't have a preference. 👍

Reply:
I would raise a PR to log a warning on 201 myself, but I can't see where the response body is bound/accessible in this file.

Reply:
Grepping the repo for "partial", it documents the 400 service behavior, but doesn't handle it. Since the SaaS products will soon return 400 instead of 201, I thought it isn't worth the effort to do anything more than this PR.

Reply:
@savage-engineer, the `WriteApiImpl` does not operate with the response HTTP body. To access the response, you should add a `next` callback https://github.com/influxdata/influxdb-client-js/blob/f129a2ca3dccedefa0042863ccf1d712196b676a/packages/core/src/results/CommunicationObserver.ts#L28

Reply:
No, if there is an error from InfluxDB, the client parses the "error state" from the response body and propagates it as an Error.